src/rules/objectLiteralShorthandRule.ts

```diff
@@ -63,6 +103,19 @@ function walk(ctx: Lint.WalkContext<void>) {
     });
 }
 
+function fixShorthandToLonghand(node: ts.ShorthandPropertyAssignment | ts.MethodDeclaration): Lint.Fix {
+    let replacementText = isMethodDeclaration(node) ? ": function" : `: ${node.name.getText()}`;
+
+    replacementText = isGeneratorFunction(node) ? `${replacementText}*` : replacementText;
+
+    const fixes: Lint.Fix = [Lint.Replacement.appendText(node.name.end, replacementText)];
+    if (isGeneratorFunction(node)) {
+        const asteriskPosition = (node as ts.MethodDeclaration).asteriskToken!.getStart();
+        fixes.push(Lint.Replacement.replaceFromTo(asteriskPosition, asteriskPosition + 1, ""));
+    }
```

Comment:
needs special handling for `async` methods as well.

before you do that, just a suggestion to simplify the fix: replace the name with the `function` keyword and insert the name plus the colon at the start of the declaration.
That way you don't have to handle the async keyword or the asterisk at all